### PR TITLE
[WIP] Migrate Hive and Iceberg Views to Trino

### PR DESCRIPTION
**Pull request by phd3**

### CODE REVIEW

**Comment by findepi** [comment]
@phd3 somehow i wasn't aware of the problem and i think it shows some gaps in our testing.
While view definition is serialized and deserialized so represents a persistent state, do we have tests ensuring cross-version compatibility?
or do we rely on our users to report problems after a release?

re: config, i'd strive to make this work out of the box; by not supporting others' views out of the box, we're putting ourselves in an inferior position.

**Comment by phd3** [comment]
@findepi 

> somehow i wasn't aware of the problem and i think it shows some gaps in our testing.
While view definition is serialized and deserialized so represents a persistent state, do we have tests ensuring cross-version compatibility?
or do we rely on our users to report problems after a release?

^ Do you mean something similar to what we have for JDBC driver? I think the view encoding and decoding has been pretty stable, so we haven't had the need for it so far. This is a one-off situation, I don't think will occur often. SQL support changes may have an impact here, but that'd have more general impact than just views. But I agree it'll be good to have a set of such tests.

> re: config, i'd strive to make this work out of the box; by not supporting others' views out of the box, we're putting ourselves in an inferior position.

^ To support "presto" views permanently (and views created before this change goes in) out of the box, I think we'll likely need to have _some_ mention of "presto" in the code or default config. Not sure if that's allowed. 

While presto -> trino view migration is better to do in bulk (by some pipeline controlled by admins), I was also thinking about adding a hive connector procedure that provides users a way to replace their views. Any thoughts on that?

**Comment by findepi** [comment]
@phd3 i think this is not the first time view definition changed unwittingly (but i may be wrong here). The problem is that persistent shape is implicitly derived from an annotated class, so it is easy to make a mistake.

> something similar to what we have for JDBC driver? 

that would be great!

or, if we could extract a view definition from the past, effectively _stubbing_ a previous version

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveConfig.java [hunk 1]
```diff
@@ -126,6 +126,9 @@
     private Duration fileStatusCacheExpireAfterWrite = new Duration(1, MINUTES);
     private long fileStatusCacheMaxSize = 1000 * 1000;
     private List<String> fileStatusCacheTables = ImmutableList.of();
+
+    // TODO: Change this to Optional.empty() along with other such Trino migration changes
+    private Optional<String> alternateViewCodecTag = Optional.of("Presto");
```
Comment (findepi) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveConfig.java:131]: We should accept our own views out of the box.
I don't see why would create burden for the users.

**Comment by bitsondatadev** [comment]
:wave: @phd3 - this PR is inactive and doesn't seem to be under development, and it might already be implemented. If you'd like to continue work on this at any point in the future, feel free to re-open.

**Pull request closed without merging**